pool/app/admin/event.ml

```diff
@@ -52,12 +53,24 @@ let handle_event ~tags pool : event -> unit Lwt.t =
         ~password:(admin.password |> User.Password.to_sihl)
         (User.EmailAddress.value admin.email)
     in
-    let%lwt (_
+    let%lwt (authorizable
               : ([> `Admin ] Guard.Authorizable.t, Common.Message.error) result)
       =
       Entity_guard.Actor.to_authorizable ~ctx user
       |> Lwt_result.map_error Pool_common.Utils.with_log_error
     in
+    let%lwt _ =
+      match authorizable, admin.roles with
+      | Ok auth, Some roles ->
```

Comment:
Could that possibly turn out as an error? If yes, it should be in the command.

Also it is not checked if the email address already exists.

Reply:
The event handler itself usually doesn't have additional validation.
In this case, Sihl will through an exception when the email address already exists.

If the user already exists is done in the web handler:
https://github.com/uzh/pool/pull/56/files/08a0a4295a8d9f7e3d66b45e4908c4cecd82b886#diff-c6711b578981e5924f39c02c41c6e3b54e520ed0efb2b0ad277c2ca3f698bf9eR103